app/Http/Controllers/RoleController.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace App\Http\Controllers;
+use App\Models\Role;
+use App\Models\User;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Contracts\View\Factory;
+use Illuminate\Contracts\View\View;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use function Webmozart\Assert\Tests\StaticAnalysis\length;
+
+
+class RoleController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return Application|Factory|View
+     */
+    public function index(): View|Factory|Application
+    {
+        $roles = Role::all();
+        return view('roles.index', ['roles' => $roles]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return Application|Factory|View
+     */
+    public function create(): View|Factory|Application
+    {
+        $count = DB::table('roles')->count();
+        $role = Role::all();
+
+        $counter = $role[$count-1]->id +1;
+
+        return view('roles.create', ['counter' =>$counter]);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param Request $request
+     * @return RedirectResponse
+     */
+    public function store(Request $request): RedirectResponse
+    {
+
+        $request->validate([
+            'name' => 'required',
+            'description' => 'required',
+        ]);
+        $role = new Role();
+        $role->name = $request->name;
+        $role->description = $request->description;
+        $role->save();
```

Comment:
In der create Methode wird eine ID für die Rolle ausgerechnet, aber hier nicht genutzt. Soll das so?